src/sipreq.c

```diff
@@ -150,7 +150,8 @@ int sip_req_send(struct ua *ua, const char *method, const char *uri,
 	if (err)
 		goto out;
 
-	err = sip_auth_alloc(&sr->auth, auth_handler, ua_account(ua), true);
+	err = sip_auth_alloc(&sr->auth, uag_sip(), auth_handler,
+		ua_account(ua), true);
```

Comment:
is it possible to implement this without breaking the sip_auth_alloc API ?

Reply:
Yes, with a setter. This was also my argument at first moment.

We discussed and found that because this solves a major security problem it maybe is better that anyone who uses this API will have to fix his/her build to solve this security problem.

But if you prefer the setter, it will be also okay for us. In this case we can decide between the old behavior. So the attack still will be possible. Or if `sip` is not set, the authorization will be canceled by default in `sip_auth_authenticate()`.

Again we are discussing about "security by default" or not. ;-)

Which would you prefer?